app/plugin_api.go

```diff
@@ -34,6 +34,32 @@ func NewPluginAPI(a *App, manifest *model.Manifest) *PluginAPI {
 	}
 }
 
+func (api *PluginAPI) applyDefaultConfigValues(pluginConfig map[string]interface{}) {
+	initialConfig := make(map[string]interface{})
+
+	if api.manifest.SettingsSchema != nil {
+		for _, settings := range api.manifest.SettingsSchema.Settings {
+			initialConfig[strings.ToLower(settings.Key)] = settings.Default
+		}
+	}
+	// Checking if plugin config was specified by user,
+	// if not we just apply default values from the initial config settings and return
```

Comment:
Might we just fold this logic into the for loop before? It seems like this is just the `!exist` case.